modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -1264,76 +1264,123 @@ func (suite *KeeperTestSuite) TestChanUpgradeCancel() {
 
-	// cause the upgrade to fail on chain b so an error receipt is written.
-	// if the counterparty (chain A) upgrade sequence is less than the current sequence, (chain B)
-	// an upgrade error will be returned by chain B during ChanUpgradeTry.
-	channel := path.EndpointA.GetChannel()
-	channel.UpgradeSequence = 1
-	path.EndpointA.SetChannel(channel)
+			path.EndpointA.ChannelConfig.ProposedUpgrade.Fields.Version = mock.UpgradeVersion
+			path.EndpointB.ChannelConfig.ProposedUpgrade.Fields.Version = mock.UpgradeVersion
 
-	channel = path.EndpointB.GetChannel()
-	channel.UpgradeSequence = 2
-	path.EndpointB.SetChannel(channel)
+			suite.Require().NoError(path.EndpointA.ChanUpgradeInit())
 
-	err := path.EndpointB.ChanUpgradeTry()
-	suite.Require().NoError(err)
+			// cause the upgrade to fail on chain b so an error receipt is written.
+			// if the counterparty (chain A) upgrade sequence is less than the current sequence, (chain B)
+			// an upgrade error will be returned by chain B during ChanUpgradeTry.
+			channel := path.EndpointA.GetChannel()
+			channel.UpgradeSequence = 1
+			path.EndpointA.SetChannel(channel)
 
-	err = path.EndpointA.UpdateClient()
-	suite.Require().NoError(err)
+			channel = path.EndpointB.GetChannel()
+			channel.UpgradeSequence = 2
+			path.EndpointB.SetChannel(channel)
 
-	errorReceipt, ok := suite.chainB.GetSimApp().IBCKeeper.ChannelKeeper.GetUpgradeErrorReceipt(suite.chainB.GetContext(), path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID)
-	suite.Require().True(ok)
+			err := path.EndpointB.ChanUpgradeTry()
+			suite.Require().NoError(err)
 
-	ctx := suite.chainA.GetContext()
-	suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.WriteUpgradeCancelChannel(ctx, path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, errorReceipt)
+			err = path.EndpointA.UpdateClient()
+			suite.Require().NoError(err)
 
-	channel = path.EndpointA.GetChannel()
+			errorReceipt, ok := suite.chainB.GetSimApp().IBCKeeper.ChannelKeeper.GetUpgradeErrorReceipt(suite.chainB.GetContext(), path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID)
+			suite.Require().True(ok)
 
-	// Verify that channel has been restored to previous state
-	suite.Require().Equal(types.OPEN, channel.State)
-	suite.Require().Equal(types.NOTINFLUSH, channel.FlushStatus)
-	suite.Require().Equal(mock.Version, channel.Version)
-	suite.Require().Equal(errorReceipt.Sequence, channel.UpgradeSequence)
+			ctx := suite.chainA.GetContext()
+			tc.malleate()
 
-	// Assert that state stored for upgrade has been deleted
-	upgrade, found := suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.GetUpgrade(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
-	suite.Require().Equal(types.Upgrade{}, upgrade)
-	suite.Require().False(found)
+			if tc.expPanic {
+				defer func() {
+					r := recover()
+					suite.NotNil(r, "The code did not panic")
+					suite.Equal(r, tc.messagePanic)
+				}()
+			}
```

Comment:
There's a built in mechanism we can use to do this

```golang
suite.Require().Panics(...)
```

I think by using this we can remove the need to rely on a specific error string as well

Reply:
thank @chatton